tests/conftest.py

```diff
@@ -396,3 +402,117 @@ def log_cli_level(pytestconfig):
 
     """
     return pytestconfig.getini('log_cli_level') or 'DEBUG'
+
+
+@pytest.fixture(scope="session")
+def run_io_in_background(request):
+    """
+    Run IO during the test execution
+    """
+    if config.RUN['cli_params'].get('io_in_bg'):
+        log.info(f"Tests will be running while IO is in the background")
+
+        request.results = list()
+        request.temp_file = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='test_status', delete=False
+        )
+
+        def get_test_status():
+            with open(request.temp_file.name, 'r') as t_file:
+                return t_file.readline()
+
+        def set_test_status(status):
+            with open(request.temp_file.name, 'w') as t_file:
+                t_file.writelines(status)
+
+        set_test_status('running')
+        request.g_sheet = GoogleSpreadSheetAPI("IO BG results", 0)
+
+        def finalizer():
+            """
+            Delete the resources created during setup, used for
+            running IO in the test background
+            """
+            set_test_status('finished')
+            try:
+                for status in TimeoutSampler(90, 3, get_test_status):
+                    if status == 'terminated':
+                        break
+            except TimeoutExpiredError:
+                log.warning(
+                    "Background IO was still in progress before IO "
+                    "thread termination"
+                )
+            if hasattr(request, 'thread'):
+                request.thread.join()
+
+            log.info(f"Background IO has stopped")
+            for result in request.results:
+                log.info(f"IOPs after FIO for pod {request.pod_obj.name}:")
```

Comment:
I missed this detail 1st time: why are you pushing pod_obj to the request object?

Reply:
no need for the request assignments. Will be removed